internal/pkg/endpoints/kubernetes.go

```diff
@@ -249,6 +249,20 @@ func getPort(o metav1.Object) string {
 	return ""
 }
 
+// parsePath parses a partial URL query from the prometheus.io/path annotation, such as `/metrics?format=foo` and
+// returns separately the path and the query. This is needed because the `prometheus.io/path` annotation is often
+// abused, and query arguments are included in it.
+// If the contents of prometheus.io/path cannot be parsed as a URL, it returns an error, which will skip the target.
+// Note that url.Parse will not return an error for an empty rawUrl.
+func parsePath(pathQuery string) (string, string, error) {
+	u, err := url.Parse(pathQuery)
+	if err != nil {
+		return "", "", fmt.Errorf("cannot parse url from %q: %w", pathQuery, err)
+	}
+
+	return u.Path, u.RawQuery, nil
```

Comment:
you could also return u and use it as base object and add to it Host and Scheme..

But having the parsePath as builder of the url would be super weird 😓

Reply:
I thought about this, refactoring this further and having a URL builder given an object. However this was very hard given how entagled the code is now :(

The main reason I did not return the URL was to avoid having to copy it for different iterations of the for loop, dereferencing the pointer for copying would have look bad :(